app/index.js

```diff
@@ -0,0 +1,86 @@
+import express from "express";
+import pg from "pg";
+
+const server = async () => {
+  const db = new pg.Client(process.env.DATABASE_URL);
+  await db.connect();
+
+  const app = express();
+
+  app.get("/transactions-history-for-addresses", async (req, res) => {
+    if (!req.query.addresses) {
+      res.status(400).json({ message: "addresses is required in query" });
+
+      return;
+    }
+
+    const queryResult = await db.query(
+      `
+      with t as (
+        SELECT "Transaction".id,
+          "Transaction".payload,
+          "Transaction".hash,
+          "Transaction".tx_index,
+          "Transaction".is_valid,
+          "Block".height
+        FROM "StakeCredential"
+        INNER JOIN "TxCredentialRelation" ON "TxCredentialRelation".credential_id = "StakeCredential".id
+        INNER JOIN "Transaction" ON "TxCredentialRelation".tx_id = "Transaction".id
+        INNER JOIN "Block" ON "Transaction".block_id = "Block".id
+        WHERE "StakeCredential".credential = ANY ($1)
+        ORDER BY "Block".height DESC,
+          "Transaction".tx_index ASC
```

Comment:
we'll have to add a LIMIT (based on some configurable JS env variable probably) to avoid the query failing due to too large payload on addresses with tens of thousands of transactions